include/secp256k1.h

```diff
@@ -526,6 +526,37 @@ SECP256K1_API extern const secp256k1_nonce_function secp256k1_nonce_function_rfc
 /** A default safe nonce generation function (currently equal to secp256k1_nonce_function_rfc6979). */
 SECP256K1_API extern const secp256k1_nonce_function secp256k1_nonce_function_default;
 
+/** Compute commitment on the client as part of the ECDSA Anti Nonce Sidechannel Protocol.
+ *
+ * ECDSA Anti Nonce Sidechannel Protocol:
+ * 1. The host draws randomness `k2`, commits to it with sha256 and sends the commitment to the client.
+ * 2. The client commits to it's original nonce `k1` using the host commitment by calling
+ *    `secp256k1_ecdsa_anti_nonce_sidechan_client_commit`. The client sends the resulting commitment
+ *   `R1` to the host.
+ * 3. The host replies with `k2` generated in step 1.
+ * 4. The client checks that the host commitment from step 1 commits to `k2` from step 3 and signs
+ *    with `secp256k1_ecdsa_sign_nonce_tweak_add`, using the `k2` as the tweak and the host
+ *    commitment as additional noncedata and sends the signature to the host.
+ * 5. The host reconstructs the curve point `R` from the signature and verifies that `R = R1 + k2*G`.
+ *
+ *  Returns 1 on success, 0 on failure.
+ *  Args:           ctx: pointer to a context object (cannot be NULL)
+ *  Out:  client_commit: pointer to a pubkey where the clients public nonce will be
+ *                       placed. (cannot be NULL)
+ *  In:           msg32: the 32-byte message hash to be signed (cannot be NULL)
+ *             seckey32: the 32-byte secret key used for signing (cannot be NULL)
+ *              noncefp: pointer to a nonce generation function. If NULL, secp256k1_nonce_function_default is used
+ *    rand_commitment32: the 32-byte randomness commitment from the host (cannot be NULL)
+ */
+SECP256K1_API int secp256k1_ecdsa_anti_nonce_sidechan_client_commit(
+    const secp256k1_context* ctx,
+    secp256k1_pubkey *client_commit,
+    const unsigned char *msg32,
+    const unsigned char *seckey32,
+    secp256k1_nonce_function noncefp,
```

Comment:
The reason why I'm not supporting arbitrary noncefps in my PR is to avoid giving a nonce function that doesn't make use of the `noncedata`. In the Schnorr case that would lead to nonce reuse and allows the host to extract the secret key. I suspect the same is true for ECDSA.

Reply:
fixed